lib/rrule/generators/generator.rb

```diff
@@ -9,6 +9,8 @@ def initialize(context)
     end
 
     def process_timeset(date, timeset)
+      return date if timeset.blank?
```

Comment:
It looks like this is always called from within a `.flat_map`, but it feels weird that the method can now return a `Date` or an `Array`.

Reply:
Good point. I did not notice. I'll take a look at that next time around.